homeassistant/components/withings/api.py

```diff
@@ -65,7 +65,7 @@ def _request(
         )
         return response.json()
 
-    async def _do_retry(self, func, attempts=3) -> Any:
+    async def _do_retry(self, func: Callable, attempts=3) -> Any:
```

Comment:
This seems like it would be better as a decorator

Reply:
Definetly, but I am planning to move this to a new lib. This file is just a stop in between of the major rework